Aleksey-Timoschenko/nodejs-aws-shop-be
# [Task 5]: Implement import-service

Within this branch:
1. New service `import-service` was implemented
2. New s3 bucket with `uploaded` and `parsed` folders was created 
3. New aws cdk stack was created
4. New lambda function `importProductsFile` was implemented. Lambda returns signed url which can be used for uploading csv files to `uploaded` folder
5. New lambda function `importFileParser`was implemented. Lambda is triggered by `s3:ObjectCreated:*` event and do the next things:
- parse csv file with csv-parser and log chunks 
- in the end move file from `uploaded` folder to `parsed` folder and remove it from `uploaded` folder
6. Unit tests for both lambdas were created

Links:
- URL for getting signed url (do not forget to provide name as query string param) https://f3izpbp52d.execute-api.eu-north-1.amazonaws.com/import 
- FE Client - https://d1akx69qkmi8fp.cloudfront.net/
- FE PR - https://github.com/Aleksey-Timoschenko/nodejs-aws-shop-react/pull/4

To verify: 
- You can upload csv file via FE client 
- OR check this video with demo - https://www.veed.io/view/d8686773-125a-4e29-aa90-50093111c89e?panel=share
